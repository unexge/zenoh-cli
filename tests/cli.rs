use std::process::Command;

use insta_cmd::{assert_cmd_snapshot, get_cargo_bin};

mod zenoht;

fn cli() -> Command {
    Command::new(get_cargo_bin("zenoh-cli"))
}

#[test]
fn test_getting_a_non_existent_value() {
    let _session = zenoht::builder()
        .add_storage(
            "test_getting_a_non_existent_value",
            zenoht::Storage::empty(),
        )
        .start();

    assert_cmd_snapshot!(cli().args(["get", "test_getting_a_non_existent_value/foo"]));
}

#[test]
fn test_getting_a_value() {
    let _session = zenoht::builder()
        .add_storage(
            "test_getting_a_value",
            zenoht::Storage::with_entries(&[("foo", "bar")]),
        )
        .start();

    assert_cmd_snapshot!(cli().args(["get", "test_getting_a_value/foo"]));
}
