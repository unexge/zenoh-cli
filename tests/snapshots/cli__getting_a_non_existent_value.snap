---
source: tests/cli.rs
info:
  program: zenoh-cli
  args:
    - get
    - foo
---
success: true
exit_code: 0
----- stdout -----
no replies received

----- stderr -----
